File: Demo_Objective-C_Storyboard/LGSideMenuControllerDemo/MainViewController.m
//
//  MainViewController.m
//  LGSideMenuControllerDemo
//
//  Created by Grigory Lutkov on 25.04.15.
//  Copyright © 2015 Grigory Lutkov <Friend.LGA@gmail.com>. All rights reserved.
//

#import "MainViewController.h"
#import "ViewController.h"
#import "LeftViewController.h"
#import "RightViewController.h"
#import "AppDelegate.h"

@interface MainViewController ()

@property (strong, nonatomic) LeftViewController *leftViewController;
@property (strong, nonatomic) RightViewController *rightViewController;
@property (assign, nonatomic) NSUInteger type;

@end

@implementation MainViewController

- (void)setupWithPresentationStyle:(LGSideMenuPresentationStyle)style type:(NSUInteger)type {
    self.leftViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"LeftViewController"];
    self.rightViewController = [self.storyboard instantiateViewControllerWithIdentifier:@"RightViewController"];

    // -----

    if (type == 0) {
        [self setLeftViewEnabledWithWidth:250.0
                        presentationStyle:style
                     alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.leftViewStatusBarStyle = UIStatusBarStyleDefault;
        self.leftViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnNone;

        // -----

        [self setRightViewEnabledWithWidth:100.0
                         presentationStyle:style
                      alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.rightViewStatusBarStyle = UIStatusBarStyleDefault;
        self.rightViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnNone;

        // -----

        switch (style) {
            case LGSideMenuPresentationStyleSlideAbove: {
                self.leftViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.9];
                self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
                self.leftViewController.tintColor = [UIColor blackColor];

                self.rightViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.9];
                self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
                self.rightViewController.tintColor = [UIColor blackColor];
                
                break;
            }
            case LGSideMenuPresentationStyleScaleFromBig:
            case LGSideMenuPresentationStyleSlideBelow:
            case LGSideMenuPresentationStyleScaleFromLittle: {
                self.leftViewBackgroundImage = [UIImage imageNamed:@"image"];
                self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
                self.leftViewController.tintColor = [UIColor whiteColor];

                self.rightViewBackgroundImage = [UIImage imageNamed:@"image2"];
                self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
                self.rightViewController.tintColor = [UIColor whiteColor];

                break;
            }
        }
    }
    else if (type == 1) {
        [self setLeftViewEnabledWithWidth:250.0
                        presentationStyle:style
                     alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnPhoneLandscape|LGSideMenuAlwaysVisibleOnPadLandscape];

        self.leftViewStatusBarStyle = UIStatusBarStyleDefault;
        self.leftViewStatusBarVisibleOptions = LGSideMenuAlwaysVisibleOnPadLandscape;
        self.leftViewBackgroundImage = [UIImage imageNamed:@"image"];

        self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
        self.leftViewController.tintColor = [UIColor whiteColor];

        // -----

        [self setRightViewEnabledWithWidth:100.0
                         presentationStyle:LGSideMenuPresentationStyleSlideAbove
                      alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.rightViewStatusBarStyle = UIStatusBarStyleDefault;
        self.rightViewStatusBarVisibleOptions = LGSideMenuAlwaysVisibleOnPadLandscape;
        self.rightViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.9];

        self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
        self.rightViewController.tintColor = [UIColor blackColor];
    }
    else if (type == 2) {
        [self setLeftViewEnabledWithWidth:250.0
                        presentationStyle:style
                     alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.leftViewStatusBarStyle = UIStatusBarStyleDefault;
        self.leftViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnAll;
        self.leftViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.9];

        self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
        self.leftViewController.tintColor = [UIColor blackColor];

        // -----

        [self setRightViewEnabledWithWidth:100.0
                         presentationStyle:style
                      alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.rightViewStatusBarStyle = UIStatusBarStyleDefault;
        self.rightViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnAll;
        self.rightViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.9];

        self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
        self.rightViewController.tintColor = [UIColor blackColor];
    }
    else if (type == 3) {
        [self setLeftViewEnabledWithWidth:250.0
                        presentationStyle:style
                     alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.leftViewStatusBarStyle = UIStatusBarStyleLightContent;
        self.leftViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnAll;
        self.leftViewBackgroundColor = [UIColor colorWithWhite:0.0 alpha:0.5];

        self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
        self.leftViewController.tintColor = [UIColor whiteColor];

        // -----

        [self setRightViewEnabledWithWidth:100.0
                         presentationStyle:style
                      alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.rightViewStatusBarStyle = UIStatusBarStyleLightContent;
        self.rightViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnAll;
        self.rightViewBackgroundColor = [UIColor colorWithWhite:0.0 alpha:0.5];

        self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
        self.rightViewController.tintColor = [UIColor whiteColor];
    }
    else if (type == 4) {
        self.swipeGestureArea = LGSideMenuSwipeGestureAreaFull;
        self.rootViewCoverColorForLeftView = [UIColor colorWithRed:0.0 green:1.0 blue:0.5 alpha:0.3];
        self.rootViewScaleForLeftView = 0.6;
        self.rootViewLayerBorderWidth = 3.0;
        self.rootViewLayerBorderColor = [UIColor whiteColor];
        self.rootViewLayerShadowRadius = 10.0;
        self.rootViewCoverColorForRightView = [UIColor colorWithRed:0.0 green:0.5 blue:1.0 alpha:0.3];

        // -----

        [self setLeftViewEnabledWithWidth:250.0
                        presentationStyle:LGSideMenuPresentationStyleScaleFromBig
                     alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.leftViewAnimationSpeed = 0.4;
        self.leftViewStatusBarStyle = UIStatusBarStyleDefault;
        self.leftViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnNone;
        self.leftViewBackgroundImage = [UIImage imageNamed:@"image"];
        self.leftViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnPadLandscape;
        self.leftViewBackgroundImageInitialScale = 1.5;
        self.leftViewInititialOffsetX = -200.0;
        self.leftViewInititialScale = 1.5;

        self.leftViewController.tableView.backgroundColor = [UIColor clearColor];
        self.leftViewController.tintColor = [UIColor whiteColor];

        // -----

        [self setRightViewEnabledWithWidth:100.0
                         presentationStyle:LGSideMenuPresentationStyleSlideAbove
                      alwaysVisibleOptions:LGSideMenuAlwaysVisibleOnNone];

        self.rightViewAnimationSpeed = 0.3;
        self.rightViewStatusBarStyle = UIStatusBarStyleDefault;
        self.rightViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnNone;
        self.rightViewBackgroundColor = [UIColor colorWithWhite:1.0 alpha:0.7];
        self.rightViewStatusBarVisibleOptions = LGSideMenuStatusBarVisibleOnPadLandscape;
        self.rightViewLayerBorderWidth = 3.0;
        self.rightViewLayerBorderColor = [UIColor blackColor];
        self.rightViewLayerShadowRadius = 10.0;

        self.rightViewController.tableView.backgroundColor = [UIColor clearColor];
        self.rightViewController.tintColor = [UIColor blackColor];
    }

    // -----

    [self.leftViewController.tableView reloadData];
    [self.leftView addSubview:self.leftViewController.tableView];

    [self.rightViewController.tableView reloadData];
    [self.rightView addSubview:self.rightViewController.tableView];
}

- (void)leftViewWillLayoutSubviewsWithSize:(CGSize)size {
    [super leftViewWillLayoutSubviewsWithSize:size];

    if (![UIApplication sharedApplication].isStatusBarHidden && (self.type == 2 || self.type == 3)) {
        self.leftViewController.tableView.frame = CGRectMake(0.0, 20.0, size.width, size.height-20.0);
    }
    else {
        self.leftViewController.tableView.frame = CGRectMake(0.0, 0.0, size.width, size.height);
    }
}

- (void)rightViewWillLayoutSubviewsWithSize:(CGSize)size {
    [super rightViewWillLayoutSubviewsWithSize:size];
    
    if (![UIApplication sharedApplication].isStatusBarHidden && (self.type == 2 || self.type == 3)) {
        self.rightViewController.tableView.frame = CGRectMake(0.0, 20.0, size.width, size.height-20.0);
    }
    else {
        self.rightViewController.tableView.frame = CGRectMake(0.0, 0.0, size.width, size.height);
    }
}

@end

File: Demo Swift/LGSideMenuControllerDemo/MainViewController.swift
//
//  MainViewController.swift
//  LGSideMenuControllerDemo
//
//  Created by Cole Dunsby on 2016-07-24.
//  Copyright © 2016 Cole Dunsby. All rights reserved.
//

import UIKit

class MainViewController: LGSideMenuController {

    private lazy var leftViewController: LeftViewController = {
        return self.storyboard?.instantiateViewControllerWithIdentifier("LeftViewController") as! LeftViewController
    }()
    
    private lazy var rightViewController: RightViewController = {
        return self.storyboard?.instantiateViewControllerWithIdentifier("RightViewController") as! RightViewController
    }()
    
    private var type: Int?
    
    func setup(presentationStyle style: LGSideMenuPresentationStyle, type: Int) {
        self.type = type
        
        if type == 0 {
            setLeftViewEnabledWithWidth(250, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            leftViewStatusBarStyle = .Default
            leftViewStatusBarVisibleOptions = .OnNone
            leftViewController.tableView.backgroundColor = .clearColor()
            
            // -----
            
            setRightViewEnabledWithWidth(100, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            rightViewStatusBarStyle = .Default
            rightViewStatusBarVisibleOptions = .OnNone
            rightViewController.tableView.backgroundColor = .clearColor()
            
            switch style {
            case .SlideAbove:
                leftViewBackgroundColor = UIColor(white: 1.0, alpha: 0.9)
                leftViewController.tintColor = .blackColor()
                
                // -----
                
                rightViewBackgroundColor = UIColor(white: 1.0, alpha: 0.9)
                rightViewController.tintColor = .blackColor()
            case .SlideBelow, .ScaleFromBig, .ScaleFromLittle:
                leftViewBackgroundImage = UIImage(named: "image")
                leftViewController.tintColor = .whiteColor()
                
                // -----
                
                rightViewBackgroundImage = UIImage(named: "image2")
                rightViewController.tintColor = .whiteColor()
            }
        } else if type == 1 {
            setLeftViewEnabledWithWidth(250, presentationStyle: style, alwaysVisibleOptions: [.OnPhoneLandscape, .OnPadLandscape])
            
            leftViewStatusBarStyle = .Default
            leftViewStatusBarVisibleOptions = .OnPadLandscape
            leftViewBackgroundImage = UIImage(named: "image")
            
            leftViewController.tableView.backgroundColor = .clearColor()
            leftViewController.tintColor = .whiteColor()
            
            // -----
            
            setRightViewEnabledWithWidth(100, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            rightViewStatusBarStyle = .Default
            rightViewStatusBarVisibleOptions = .OnPadLandscape
            rightViewBackgroundColor = UIColor(white: 1.0, alpha: 0.9)
            
            rightViewController.tableView.backgroundColor = .clearColor()
            rightViewController.tintColor = .blackColor()
        } else if type == 2 {
            setLeftViewEnabledWithWidth(250, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            leftViewStatusBarStyle = .Default
            leftViewStatusBarVisibleOptions = .OnAll
            leftViewBackgroundColor = UIColor(white: 1.0, alpha: 0.9)
            
            leftViewController.tableView.backgroundColor = .clearColor()
            leftViewController.tintColor = .blackColor()
            
            // -----
            
            setRightViewEnabledWithWidth(100, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            rightViewStatusBarStyle = .Default
            rightViewStatusBarVisibleOptions = .OnAll
            rightViewBackgroundColor = UIColor(white: 1.0, alpha: 0.9)
            
            rightViewController.tableView.backgroundColor = .clearColor()
            rightViewController.tintColor = .blackColor()
        } else if type == 3 {
            setLeftViewEnabledWithWidth(250, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            leftViewStatusBarStyle = .LightContent
            leftViewStatusBarVisibleOptions = .OnAll
            leftViewBackgroundColor = UIColor(white: 0.0, alpha: 0.5)
            
            leftViewController.tableView.backgroundColor = .clearColor()
            leftViewController.tintColor = .whiteColor()
            
            // -----
            
            setRightViewEnabledWithWidth(100, presentationStyle: style, alwaysVisibleOptions: .OnNone)
            
            rightViewStatusBarStyle = .LightContent
            rightViewStatusBarVisibleOptions = .OnAll
            rightViewBackgroundColor = UIColor(white: 0.0, alpha: 0.5)
            
            rightViewController.tableView.backgroundColor = .clearColor()
            rightViewController.tintColor = .whiteColor()
        } else if type == 4 {
            swipeGestureArea = .Full
            rootViewCoverColorForLeftView = UIColor(red: 0.0, green: 1.0, blue: 0.5, alpha: 0.3)
            rootViewScaleForLeftView = 0.6
            rootViewLayerBorderWidth = 3
            rootViewLayerBorderColor = .whiteColor()
            rootViewLayerShadowRadius = 10
            rootViewCoverColorForRightView = UIColor(red: 0.0, green: 0.5, blue: 1.0, alpha: 0.3)
            
            // -----
            
            setLeftViewEnabledWithWidth(250, presentationStyle: .ScaleFromBig, alwaysVisibleOptions: .OnNone)
            
            leftViewAnimationSpeed = 0.4
            leftViewStatusBarStyle = .Default
            leftViewStatusBarVisibleOptions = .OnNone
            leftViewBackgroundImage = UIImage(named: "image")
            leftViewStatusBarVisibleOptions = .OnPadLandscape
            leftViewBackgroundImageInitialScale = 1.5
            leftViewInititialOffsetX = -200
            leftViewInititialScale = 1.5
            
            leftViewController.tableView.backgroundColor = .clearColor()
            leftViewController.tintColor = .whiteColor()
            
            // -----
            
            setRightViewEnabledWithWidth(100, presentationStyle: .SlideAbove, alwaysVisibleOptions: .OnNone)
            
            rightViewAnimationSpeed = 0.3
            rightViewStatusBarStyle = .Default
            rightViewStatusBarVisibleOptions = .OnNone
            rightViewBackgroundColor = UIColor(white: 1.0, alpha: 0.7)
            rightViewStatusBarVisibleOptions = .OnPadLandscape
            rightViewLayerBorderWidth = 3
            rightViewLayerBorderColor = .blackColor()
            rightViewLayerShadowRadius = 10
            
            rightViewController.tableView.backgroundColor = .clearColor()
            rightViewController.tintColor = .blackColor()
        }
        
        // -----
        
        leftViewController.tableView.reloadData()
        leftView().addSubview(leftViewController.tableView)
        
        rightViewController.tableView.reloadData()
        rightView().addSubview(rightViewController.tableView)
    }
    
    override func leftViewWillLayoutSubviewsWithSize(size: CGSize) {
        super.leftViewWillLayoutSubviewsWithSize(size)
        
        if UIApplication.sharedApplication().statusBarHidden && (type == 2 || type == 3) {
            leftViewController.tableView.frame = CGRect(x: 0, y: 20, width: size.width, height: size.height - 20)
        } else {
            leftViewController.tableView.frame = CGRect(origin: CGPoint.zero, size: size)
        }
    }
    
    override func rightViewWillLayoutSubviewsWithSize(size: CGSize) {
        super.rightViewWillLayoutSubviewsWithSize(size)
        
        if UIApplication.sharedApplication().statusBarHidden && (type == 2 || type == 3) {
            rightViewController.tableView.frame = CGRect(x: 0, y: 20, width: size.width, height: size.height - 20)
        } else {
            rightViewController.tableView.frame = CGRect(origin: CGPoint.zero, size: size)
        }
    }

}
